tests/test_builtins.py

```diff
@@ -239,6 +239,72 @@ def test_if_parallel(self):
             assert np.all(a[::-1, 3:6] - np.array(f.data[12:18, 9:12]) == 0)
             assert np.all(a[3:6, ::-1] - np.array(f.data[9:12, 12:18]) == 0)
 
+    @pytest.mark.parametrize('ndim, nbl', [
+        (2, 0), (2, 3), (3, 0), (3, 3)
+    ])
+    def test_if_halo(self, ndim, nbl):
+        """
+        Test that FD halo is padded as well.
+        """
+        grid = Grid(tuple([11]*ndim))
+        f = Function(name="f", grid=grid)
+        a = np.zeros(tuple([11-2*nbl]*ndim))
+        a[..., 0] = 1
+        a[..., -1] = 3
+        if ndim == 3:
+            a[:, 0, :] = 5
+            a[:, -1, :] = 6
+        a[0, ...] = 2
+        a[-1, ...] = 4
+
+        initialize_function(f, a, nbl)
+
+        assert np.all(np.take(f._data_with_outhalo, 0, axis=0) == 2)
+        assert np.all(np.take(f._data_with_outhalo, -1, axis=0) == 4)
+        if ndim == 3:
+            assert f._data_with_outhalo[7, 7, 7] == 0
+            assert np.take(f._data_with_outhalo, 0, axis=-1)[7, 7] == 1
+            assert np.take(f._data_with_outhalo, -1, axis=-1)[7, 7] == 3
+            assert np.take(f._data_with_outhalo, 0, axis=1)[7, 7] == 5
+            assert np.take(f._data_with_outhalo, -1, axis=1)[7, 7] == 6
+        else:
+            assert f._data_with_outhalo[7, 7] == 0
+            assert np.take(f._data_with_outhalo, 0, axis=-1)[7] == 1
+            assert np.take(f._data_with_outhalo, -1, axis=-1)[7] == 3
+
+    @skipif('nompi')
+    @pytest.mark.parametrize('nbl', [0, 2])
+    @pytest.mark.parallel(mode=4)
+    def test_if_halo_mpi(self, nbl):
```

Comment:
I don't know why, but it looks like my comment about merging this and the non-MPI test into one test with `pytest.mark.parallel(mode=[1, 2, 4])` has disappeared?

Reply:
`mode=1` is still MPI with one rank, not non-mpi.